### ISSUE TITLE
Support: formatting issues.

### ISSUE THREAD

**Issue by lebarondemerde** [irssi/irssi#747]
Hello, I am trying to create a irssi theme but since days I cannot figure out how to accomplish one thing.

The theme has the nicknames aligned. I am trying to do the same with the others messages without using extra/external thing. Looking around I find something that works in the [weed](https://github.com/ronilaukkarinen/weed/blob/master/weed.theme) theme but feel more like a band-aid.

The problem are two:

- how to proper align those messages
- how to selectively change the "line_start" color.

In other worlds I am trying to transform this:
`join = "                          %g*%K | %K{channick_hilight $0} %gjoined%K {channel $2}";`

in something more like this:
`pubmsg = "{pubmsgnick {pubnick $[-16]0}$2}$1";`

![2017-09-06-202543_1600x900_scrot](https://user-images.githubusercontent.com/20615335/30231971-f5cc397c-94dc-11e7-99c5-5a52a4bce0c9.png)

EDIT: The perfect format would be something like:

_color_ * _space_ nickname _color_ | message _color_ message message

I already tried #irssi but it seem no one could help.

[THIS](https://pastebin.com/yuu5hvh6) is the complete (un-finished) theme.

_Sorry, it seem more like a support ticket than an issue but I really do not know if what I am trying to accomplish is currently possible._

Thanks!

**Comment by lebarondemerde** [comment]
Well, Ihad a small advance:
`join = "%w{channick_hilight $[-14]0}  %g| %gjoined %w{channel $2}";`

**Comment by ailin-nemui** [comment]
I don't understand what exactly you are looking for? A way to avoid having to use many spaces, on certain messages?

**Comment by lebarondemerde** [comment]
Hi @ailin-nemui !

I am trying to create a theme where the user can quick identify what is going on the channel. This is the main goal.

The basic idea is to keep user related actions right aligned and the others messages not. Use of the colors with purpose to help with the identification, and at same time avoiding to make the things too colorful. Still it should work for B&W terminals, but not tested yet. 

Also, do not rely on external resources, like scripts, but just on the native capabilities. In the same way, I want to keep the configuration file as easy to understand as possible.

That said, the basic differences between these lines (below) are (aside of the lack if the "*" in the second) are:

`join = " %g*%K | %K{channick_hilight $0} %gjoined%K {channel $2}";`
`join = "%w{channick_hilight $[-14]0} %g| %g joined %K{channel $2}";`

To align the first, many manual spaces would be needed what would not make it clear/easy/quick to understand and adjust. Also, (IIRC) if the text is too long that cause a line break, the second line will not be indented.

So, this particular point I managed to figure out.

The second is: many messages do use "line_start" and I do not know if it is possible to customize it per message type, aside from leaving it "empty" and add in for each message. But a quick test I did made me think it would be very hard to proper align/configure every single message manually.

Basically, even if carefully aligning them manually, it still result in a tiny misalignment in relation of what were aligned using "$[-14]0" resource.

Basically is that. I am using irssi since about couple of weeks, and got back to irc about a couple of months, after several years without actively using it (since 1998 IIRC).

So, I am quite lost, and many configuration options I still do not have any idea what they do. Fortunately, I found the weed theme what is quite useful to enlight some options.

However, I think about 1/5 of the theme is done already, the rest are more like details.

Thank you for your reply.

**Comment by ailin-nemui** [comment]
you are right, that there is no built in way to say for example $[-14]SPACE

so, the only thing you can do there if you want to stay native is to put 14 spaces manually

on line breaks, the line will be indented at the position where you write %|

and you cannot customise line_start per message type

however, there should not be any misalignment if you are careful :-)

**Comment by lebarondemerde** [comment]
@ailin-nemui 

Got it.

Thank you!
